js/app.js

```diff
@@ -0,0 +1,131 @@
+
+let arrayWithArray = [["firstElement", "secondElement"], "second element", "third element"]
+
+console.log(arrayWithArray[0][1]);
+
+// Create an object that has a property that is an object. Log one of the properties of that inner object.
+
+let objectWithObject = {
+  firstElement: {
+                firstInnerElement: "firstInnerValue",
+                secondInnerElement: "secondInnerValue"
+                },
+  secondElement: "Second Value"
+}
+
+console.log(objectWithObject.firstElement.secondInnerElement)
+
+
+//Create a function that returns an array. Log an element of the array.
+
+let returnArray = () => {
+  return ["firstElement", "secondElement", "thirdElement"]
+}
+console.log(returnArray()[2])
+
+// Create a function that returns a function. Call that inner function
+
+let functionWithInnerFunction = () => {
+  let innerFunction = () => {
+    console.log("innerFunction output");
+  }
+  return innerFunction();
+};
+functionWithInnerFunction()
+
+// Create an array that has a function that returns a function. Call the inner function.
+
+let arrayWithFunction = ["firstElement", function functionWithInnerFunction() {
+  let innerFunction = () => {
+    console.log("innerFunction in a function in an array output");
+  }
+  return innerFunction();
+}, "third element"]
+
+arrayWithFunction[1]();
+
+//Define two functions and set them to variables
+//The second function takes a parameter
+//Call the second function, passing in the variable that references the first function as the parameter
+//In the definition of the second function, invoke (call) the parameter that is being passed into it. Remember, this parameter is a function
+
+
+let function1 = () => {
+  console.log("function callback success");
+}
+let function2 = (param) => {
+  return param();
+}
+function2(function1);
+
+//Correctly indent the following code:
+
+if(true){
+    const a = 2 + 2;
+    console.log(a);
+        }
+if(true){
+  if(false){
+    console.log('hi');
+  }
+}
+
+//Semantic naming of variables
+
+//Fix this variable to have a better name:
+
+const arrayOfEvens = [2, 4, 6, 8, 10];
+
+//Function definition placement
+
+//Clean up this code, so that it works and has function definitions in the correct place
+
+
+
+const bar = () => {
+    console.log('bar here');
+}
+bar();
+
+const foo = () => {
+    console.log('foo here');
+}
+foo();
+
+//Write your own comments for each line of code:
+
+// function adds two numbers, which are the parameters
+const addTwoNums = (firstNum, secondNum)=>{
+    // declares answer to be returned, which is the sum of the two parameters
+    const finalValue = firstNum + secondNum;
+    // returns the answer
+return finalValue;
+}
+/*
+foo();
+
+const foo ()=>{
+    console.log('hi');
+}
+*/
+
+// the error means that the anonymous function is not hoisted up to the function call
+
+const b = 5;
+
+if(b === 5){ //will be false
+    console.log('this line should execute');
+}
+
+const a = 5;
```

Comment:
....but then you went and ignored the stipulation in the second part!! (change only the line that has the console.log on it)...